demo/src/App.js

```diff
@@ -1,80 +1,50 @@
-import React, { PureComponent } from 'react';
-import Slider from '@material-ui/core/Slider';
-import { withStyles } from '@material-ui/core/styles';
-import { MicroscopyViewer } from '../../src';
-import { source } from './source-info';
-import './App.css';
-
-const RedSlider = withStyles({
-  root: {
-    color: 'red'
-  }
-})(Slider);
-const BlueSlider = withStyles({
-  root: {
-    color: 'blue'
-  }
-})(Slider);
-const GreenSlider = withStyles({
-  root: {
-    color: 'green'
-  }
-})(Slider);
-const OrangeSlider = withStyles({
-  root: {
-    color: 'orange'
-  }
-})(Slider);
+import React, { PureComponent } from "react";
+import Slider from "@material-ui/core/Slider";
+import { withStyles } from "@material-ui/core/styles";
+import { MicroscopyViewer } from "../../src";
+import { source } from "./source-info";
+import "./App.css";
 
 export default class App extends PureComponent {
   constructor(props) {
     super(props);
-    this.handleRedSliderChange = this.handleRedSliderChange.bind(this);
-    this.handleGreenSliderChange = this.handleGreenSliderChange.bind(this);
-    this.handleBlueSliderChange = this.handleBlueSliderChange.bind(this);
-    this.handleOrangeSliderChange = this.handleOrangeSliderChange.bind(this);
     this.resize = this.resize.bind(this);
+    var sliderValues = {};
+    var colorValues = {};
+    var sliders = [];
+    const colorOptions = [
+      [255, 0, 0],
+      [0, 255, 0],
+      [0, 0, 255],
+      [255, 128, 0]
+    ];
+    Object.keys(source.channels).forEach((channel, i) => {
+      const sliderObj = {};
+      sliderValues[channel] = [0, 20000];
+      colorValues[channel] = colorOptions[i];
+      sliderObj[channel] = withStyles({
+        root: {
+          color: `rgb(${colorOptions[i]})`
+        }
+      })(Slider);
+      sliders.push(sliderObj);
+    });
```

Comment:
In this case, just use `map` instead of `forEach` to set the value of `sliders` as a const.

Reply:
I prefer having one iteration that sets all of these values.  Is it ok to leave?